src/python_testing/TC_SU_4_1.py

```diff
@@ -0,0 +1,406 @@
+        asserts.assert_equal(
+            fabric2_val[0],
+            provider_th3_for_fabric2,
+            "Fabric 2 provider should remain TH3 after ConstraintError on Fabric 1"
+        )
+        logger.info(f'Step #5 - Fabric 2 DefaultOTAProviders unchanged as expected: {fabric2_val}')
+
+        self.step(6)
+        # Create Empty Providers list
+        providers_list_empty = []
+        logger.info(f'Step #6 - Providers list updated with provider empty: {providers_list_empty}')
+
+        # Update attribute with empty providers list (TH3 for fabric 2)
+        attr = self.cluster_otar.Attributes.DefaultOTAProviders(value=providers_list_empty)
+
+        # Write updated DefaultOTAProviders attribute to DUT using TH3 (Fabric 2)
+        resp = await th3.WriteAttribute(
+            nodeId=self.dut_node_id,
+            attributes=[(0, attr)],
+        )
+        logger.info(f'Step #6- Write DefaultOTAProviders response on TH3 Fabric 2: {resp}')
+        asserts.assert_equal(resp[0].Status, Status.Success, "Failed to write DefaultOTAProviders attribute")
+        logger.info("Step #6 - Write operation correctly performed with empty list")
+
+        # Verify DefaultOTAProviders attribute on the DUT after write (TH3 on Fabric 2)
+        th3_actual_otap_info = await self.read_single_attribute_check_success(
+            dev_ctrl=th3,
+            cluster=self.cluster_otar,
+            attribute=self.cluster_otar.Attributes.DefaultOTAProviders)
+
+        actual_provider = th3_actual_otap_info
+
+        # Verify DefaultOTAProviders attribute on Fabric 2 is empty
+        asserts.assert_true(len(actual_provider) == 0, "DefaultOTAProviders list is not empty")
+        asserts.assert_equal(actual_provider, [], "DefaultOTAProviders on TH3 (Fabric 2) should be empty")
+        logger.info(f'Step #6 - Fabric 2 DefaultOTAProviders attribute is empty as expected: {actual_provider}')
+
+        # NOTE:
+        # - Writing an empty list succeeds and clears the providers for that fabric.
+        # - Other fabrics are not affected.
+
+        # Verify DefaultOTAProviders attribute on Fabric 1 after write is still TH4 (using TH on Fabric 1)
+        th_actual_otap_info = await self.read_single_attribute_check_success(
+            dev_ctrl=th,
+            cluster=self.cluster_otar,
+            attribute=self.cluster_otar.Attributes.DefaultOTAProviders)
+        logger.info(f'Step #6 - Read DefaultOTAProviders attribute on DUT using TH (Fabric 1): {th_actual_otap_info}')
+
+        # Verify Fabric 1 still has exactly one provider
+        asserts.assert_equal(len(th_actual_otap_info), 1, "DefaultOTAProviders list on Fabric 1 should contain one provider")
+
+        # Remaining provider must be unchanged (either TH2 or TH4)
+        asserts.assert_equal(
+            th_actual_otap_info[0],
+            provider_th4_for_fabric1,
+            "DefaultOTAProviders on Fabric 1 should remain set to TH2 or TH4"
```

Comment:
This is checking that it's TH4 only though. The test plan is wrong here - it specifies TH2 or TH4 in step 5, then only TH4 in step 6